### DISCUSSION TITLE
VSCode with Imp

### DISCUSSION THREAD

**Discussion by samchouse** [arkane-systems/bottle-imp#19] [Q&A] [answered]
How would I get VSCode to launch new terminals with Imp instead of just the vanilla shell?

**Comment by cerebrate** [comment, marked as answer]
Turns out you can do this with a modification of the technique used to do this under _genie_ . It involves a little bit of a hack, but it's not that bad.

First, turn off automatically updating extensions in VS Code. (Not strictly necessary, but since you will need to patch a file every time the _WSL_ extension updates, it's just easier.)

Second, go to the `.vscode\extensions\ms-vscode-remote.remote-wsl-0.66.3\scripts` directory under your Windows user profile. The version number after _ms-vscode-remote.remote-wsl-_ will vary depending on which version of the WSL extension you have installed; you want whatever the latest one is.

In the scripts folder, there is one called _wslServer.sh_. Open it up, and after the #! line and copyright messages but before the rest of the script, insert the following:

```
if command -v imp > /dev/null; then
        loginctl session-status > /dev/null 2>&1 || exec imp -c "$0" "$@"
fi
```

Third, make sure you don't have any old instances of the VS Code server running in your WSL instance, as this change will only affect new ones.

And there you go!

**Reply by yorkshirelandscape** [reply]
I was able to convince GitHub Copilot to help me write a script to apply this patch with a cron job if necessary. If that's something you think is worth sharing here, I'd be happy to. It originally had some misgivings about mucking about with the .vscode folder, but relented when I explained the situation.

**Comment by cerebrate** [comment]
Note: at the moment, running the Windows VS Code from inside WSL with `code` has an issue, due to a _bottle-imp_ bug that will be fixed when 0.11 ships. Until then, you need to manually recreate the WSL_DISTRO_NAME variable with

```
export WSL_DISTRO_NAME=Debian
```

or insert-distro-name-here in order for VS Code to realize that it should be using the remote extension and not just opening the files via `//wsl.localhost`.

**Reply by cerebrate** [reply]
This is now fixed as promised in 0.11.
